### DISCUSSION TITLE
Persist a single Slice in a Bounded store

### DISCUSSION THREAD

**Discussion by ithilbor** [pmndrs/zustand#1630] [Q&A] [answered]
Hi I 'm trying to persist a single slice in a bounded store with multiple slices in this way:

```node
import { create, StateCreator } from "zustand";
import { createJSONStorage, persist } from "zustand/middleware";

interface BearSlice {
  bears: number
  addBear: () => void
  eatFish: () => void
}
const createBearSlice: StateCreator<
  BearSlice & FishSlice,
  [],
  [],
  BearSlice
> = (set) => ({
  bears: 0,
  addBear: () => set((state) => ({ bears: state.bears + 1 })),
  eatFish: () => set((state) => ({ fishes: state.fishes - 1 })),
})

interface FishSlice {
  fishes: number
  addFish: () => void
}
const createFishSlice: StateCreator<
  BearSlice & FishSlice,
  [],
  [],
  FishSlice
> = (set) => ({
  fishes: 0,
  addFish: () => set((state) => ({ fishes: state.fishes + 1 })),
})

const useBoundStore = create<BearSlice & FishSlice>()((...a) => ({
  ...createBearSlice(...a),
  ...persist(createFishSlice, {
    name: "fish-store",
    storage: createJSONStorage(() => sessionStorage),
  })(..a),
}))
```

But when the value of the fishes updates it also put in the persisted state the bears that is something I don't wan't into it. If i use `partialize` it works but I was wondering if there is a way of do it without `partialize`. Thanks in advance!

**Comment by ithilbor** [comment]
This is another example more similar to what I'm doing:

https://tsplay.dev/WJAylm

As you can see there is no error but when I upgrade the fishes from the `addFishAndBear` function the fishes in the store won't update and also I'm trying to use directly the function `addFishes` it puts into the store also the `bears`. Thanks again.

**Comment by dai-shi** [comment, marked as answer]
> If i use `partialize` it works but I was wondering if there is a way of do it without `partialize`.

`partialize` sounds like a right solution. Why do you want to avoid it?

btw, I would consider having two separate stores alternatively.

**Reply by ithilbor** [reply]
Because by searching in these discussions I saw that some would not suggested it as a good solution, I think because to avoid the errors that I've encountered you need to persist all the store and then by using `partialize` your saying to store only one parameters or avoid some other but the main directive is to persist everything and It can be a problem if due to a bug everything is persisted so keep it separate in two different stores sounds a to be a good idea in this way.

**Reply by ithilbor** [reply]
So if there is not another way to persist only one slice of the store I think I will go for two different stores in order to keep also the logic separated even if have everything in one store It's handy!
PS: `partialize` by the way at run time is doing some kind of filter when it comes to store the state or it works in a different way?

**Reply by dai-shi** [reply]
> PS: `partialize` by the way at run time is doing some kind of filter when it comes to store the state or it works in a different way?

Not sure if I get the question. Does the implementation detail help?

https://github.com/pmndrs/zustand/blob/c1970e9358189fbcde952fe55c7a9a4d21e93935/src/middleware/persist.ts#L381-L387

**Reply by ithilbor** [reply]
Yes, thank you so much @dai-shi !
